server/api/api.go

```diff
@@ -101,15 +116,49 @@ func (a *API) checkCSRFToken(r *http.Request) bool {
 	return false
 }
 
+func (a *API) getContainer(r *http.Request) (store.Container, error) {
+	if a.WorkspaceAuthenticator == nil {
+		// Native auth: always use root workspace
+		container := store.Container{
+			WorkspaceID: "",
+		}
+		return container, nil
+	}
+
+	vars := mux.Vars(r)
+	workspaceID := vars["workspaceID"]
+	container := store.Container{
+		WorkspaceID: workspaceID,
+	}
+
+	if workspaceID == "" || workspaceID == "0" {
```

Comment:
Yeah, native auth uses workspaceID 0 for the API calls. It's more a developer-case if you switch providers. DoesUserHaveWorkspaceAccess would fail that case anyway, but this avoids a call.